src/event-handler/index.ts

```diff
@@ -1,13 +1,35 @@
+import type {
+  EmitterAnyEvent,
+  EmitterEventName,
+  EmitterWebhookEvent,
+  HandlerFunction,
+  Options,
+  State,
+  WebhookEventHandlerError,
+} from "../types";
 import {
   receiverOn as on,
   receiverOnAny as onAny,
   receiverOnError as onError,
 } from "./on";
 import { receiverHandle as receive } from "./receive";
 import { removeListener } from "./remove-listener";
-import { Options, State } from "../types";
 
-export function createEventHandler(options: Options<any>) {
+interface EventHandler<TTransformed = unknown> {
```

Comment:
❓ Why `TTransformed` for naming?

Reply:
Should this be
```suggestion
interface EventHandler<TTransformed = {}> {
```

Considering `U = {}` here?
https://github.com/octokit/webhooks.js/blob/d3df9f3d787990b60528cce5fba8365cdaa9ce61/src/index.ts#L24

Reply:
This is the name I used for the generic in `HandlerFunction` - ideally I'd rename `U` to match, but held off doing that for now to reduce the scope of the changes in both this PR & in #406 .

The reasoning for the name is that it gives you more information about what the generic actually represents, which is good as generics tend to exist in a vacuum so it's not as easy to know what values should be being passed in when using the type (i.e "should this be the event *payload*, or the result of what my custom transformer function is doing?").

I'm not against single letter generics, but try to only use them when they're obvious i.e 

```
removeListener<E extends EmitterEventName>(
    event: E | E[],
    callback: HandlerFunction<E, TTransformed>
  ): void;
```

Because `E` has a constraint, it's far more obvious that it's the name of an event, but if `TTransformed` was just `U` you'd not have a lot of information on what it's actually representing (as the `extends` doesn't clear things up either).

> Considering U = {} here?

[`{}` is actually means "any non-nullish value", so it's not safe to use](https://github.com/typescript-eslint/typescript-eslint/issues/2063#issuecomment-675156492). `unknown` is the better type as we've got no constraints on what type can be passed, meaning a user can transform the value into *anything* (including nullish values).

I believe that `U = {}` should be changed to use `unknown` as well, but I've not done it because of scoping + it's one of those things that while I know of the gotcha I can't remember all the rules around it well enough to do the replacement without playing around with it first (which I need the time to do) :)

Reply:
I'm a big fan of your explanations. Thanks for sharing your knowledge and taking your time on giving explanations in detail.